packages/front-end/components/Experiment/VariationDataInput.tsx

```diff
@@ -1,63 +1,111 @@
-import { useFieldArray, UseFormReturn } from "react-hook-form";
-import { ExperimentInterfaceStringDates } from "back-end/types/experiment";
-import { MdDeleteForever } from "react-icons/md";
-import Field from "../Forms/Field";
+import { Variation } from "back-end/types/experiment";
+import {
+  closestCenter,
+  DndContext,
+  KeyboardSensor,
+  PointerSensor,
+  useSensor,
+  useSensors,
+} from "@dnd-kit/core";
+import {
+  arrayMove,
+  rectSortingStrategy,
+  SortableContext,
+  sortableKeyboardCoordinates,
+} from "@dnd-kit/sortable";
 import { GBAddCircle } from "../Icons";
+import { DraggableVariationData } from "./DraggableVariationData";
 
 export interface Props {
-  form: UseFormReturn<Partial<ExperimentInterfaceStringDates>>;
+  variations: Variation[];
+  setVariations?: (variations: Variation[]) => void;
   className?: string;
 }
 
-export default function VariationDataInput({ form, className = "" }: Props) {
-  const variations = useFieldArray({
-    control: form.control,
-    name: "variations",
-  });
+export default function VariationDataInput({
+  variations,
+  setVariations,
+  className = "",
+}: Props) {
+  function generateUniqueId() {
```

Comment:
In order to ensure no variations have duplicate id's, this will generate a unique id whenever a new variation is added. It will loop through all existing variations, and returns a value that is the current highest id + 1.

This resolves a bug that would surface if someone deleted a variation other than the last variation, and then added a new one. Before this function, it would result in the last two indexes of the array having duplicate id's, since they are initially based on their index in the array.